modules/Core/src/main/java/org/terasology/logic/inventory/CharacterInventorySystem.java

```diff
@@ -114,12 +114,16 @@ public void onDropItemRequest(DropItemRequest event, EntityRef character) {
             }
         }
 
-        ItemComponent itemComponent = pickupItem.getComponent(ItemComponent.class);
-        itemComponent.timeToPickUp = 2000;
-        itemComponent.pickUpTime = time.getGameTimeInMs();
-        pickupItem.saveComponent(itemComponent);
-
         pickupItem.send(new DropItemEvent(event.getNewPosition()));
+
+        if (pickupItem.hasComponent(PickupComponent.class))
+        {
+            PickupComponent pickupComponent = pickupItem.getComponent(PickupComponent.class);
+            pickupComponent.timeToPickUp = 2000;
```

Comment:
As one final thought about the hardcoded delay amount.  Could we let the itemPickup.prefab define the timeToPickUp?  That way it comes configurable out of the box. :)